### ISSUE TITLE
Les relations d'amitiés sont doublées

### ISSUE THREAD

**Issue by agourd** [valoulou/WakeMeUp#1]
Si A est amis avec B alors la table `Friends` contiens les enregistrements suivants:

id | idUser | idAmi | pending | hasAccepted
--|--|--|--|--|
1|A|B|false|true
2|B|A|false|true

C'est pas forcement très grave, mais dans l'absolu il vaut mieux éviter ce genre de situation parce que ça peut mener à des états incohérents. Tel que je le comprends dans le code, l'acceptation d'une invitation d'ami va se faire en deux temps (comprendre deux transactions SQL).

```php
$friends->idUser = $o_member["id"];
$friends->idAmi = $o_another["id"];
$found_relation = $friends->Search();

foreach ($found_relation as $o_relation) {
	$friends->id = $o_relation["id"];
	$friends->hasAccepted = "true";
	$friends->pending = "false";
	$friends->Save(); // Transaction 1
}


$friends->idUser = $o_another["id"];
$friends->idAmi = $o_member["id"];
$found_ano_relation = $friends->Search();

foreach ($found_ano_relation as $o_relation) {
	$friends->id = $o_relation["id"];
	$friends->hasAccepted = "true";
	$friends->Save(); // Transaction 2
}
```

Du coup si une des deux transactions échoue, on aura potentiellement A qui a accepté B comme ami, mais B qui n'a pas accepté A. Ce qui donnerais en base:

id | idUser | idAmi | pending | hasAccepted
--|--|--|--|--|
1|A|B|false|true
2|B|A|true|false

Si fonctionnement, il est voulu que les amitiés puissent être "unilatérale", dans ce cas pas de soucis mais il faut revoir le bout de code au dessus, sinon il faudrait revoir le fonctionnement de cette table et n'avoir qu'un enregistrement par relation d'amitié.

**Comment by valoulou** [comment]
Je suis d'accord avec toi j'avoue que c'est une partie que je n'ai pas beaucoup contrôlé. Mais c'est un bon point que tu soulève =)

On pourra en reparler demain pendant notre petite réunion.

Bien joué !
